### PR TITLE
build //client/web:bundle with NODE_ENV=production

### PR DESCRIPTION
**Pull request by sqs**
Previously, the bundle was being built with NODE_ENV=development. This caused unnecessary code to exist in the production bundle (and caused a devtools console warning for users with the React devtools browser extension).

Now, the bundle is being built correctly with NODE_ENV=production. It also uses linked sourcemaps and simple minification (not obfuscation) for the prod bundle.

## Test plan

Manually built a prod bundle then served it and ensured that there was no console log and that the sources were minified.

### CODE REVIEW

File: client/web/BUILD.bazel [hunk 1]
```diff
@@ -2140,9 +2138,8 @@ esbuild_web_app(
         ":web_lib",
         "//:package_json",
     ],
-    config = "//client/web/dev:esbuild-config",
+    config = "//client/web/dev:esbuild-config-production",
     define = {
-        "process.env.NODE_ENV": "\"production\"",
         "process.env.INTEGRATION_TESTS": "true",
```
Comment (valerybugakov) [inline client/web/BUILD.bazel:2143]: Does the problem we're solving here mean that this env variable doesn't apply either? 
Also, we probably do not want to use this for the bundle we serve in production.

File: client/web/dev/utils/environment-config.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { getEnvironmentBoolean, STATIC_ASSETS_PATH } from '@sourcegraph/build-co
 
 import { DEFAULT_SITE_CONFIG_PATH } from './constants'
 
-const NODE_ENV = process.env.NODE_ENV || (process.env.BAZEL_BIN ? 'production' : 'development')
+const NODE_ENV = process.env.NODE_ENV as 'production' | 'development'
```
Comment (valerybugakov) [inline client/web/dev/utils/environment-config.ts:11]: Should we set it to `'development` for non-production builds explicitly?

**Review by valerybugakov — APPROVED** [review]
Approving to unblock the critical fix. Thank you! 
Left a couple of questions.

File: client/web/BUILD.bazel (hunk 1 above)
Comment (sqs) [reply client/web/BUILD.bazel:2143]: This //client/web:app-bundle target is only used in the integration tests. The `define`s are indeed merged with the JS-defined esbuild config, but `process.env.NODE_ENV` in the JS process itself is not derived from this. So the deleted line was ineffective.

File: client/web/dev/utils/environment-config.ts (hunk 2 above)
Comment (sqs) [reply client/web/dev/utils/environment-config.ts:11]: yeah, i should give a default

File: client/web/BUILD.bazel (hunk 1 above)
Comment (valerybugakov) [reply client/web/BUILD.bazel:2143]: Makes sense! 👍

**Pull request merged**